src/Compilers/Core/MSBuildTask/Microsoft.Build.Tasks.CodeAnalysis.csproj

```diff
@@ -66,9 +66,8 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Build.Framework" Version="$(RefOnlyMicrosoftBuildFrameworkVersion)" ExcludeAssets="Runtime" />
     <PackageReference Include="Microsoft.Build.Tasks.Core" Version="$(RefOnlyMicrosoftBuildTasksCoreVersion)" ExcludeAssets="Runtime" />
-    <PackageReference Include="System.IO.Pipes.AccessControl" Version="$(SystemIOPipesAccessControlVersion)" Condition="'$(TargetFramework)' == 'netcoreapp3.1'" />
-    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" Condition="'$(TargetFramework)' != 'netcoreapp3.1'" />
-    <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="$(SystemRuntimeCompilerServicesUnsafeVersion)" Condition="'$(TargetFramework)' != 'netcoreapp3.1'" />
+    <PackageReference Include="System.Memory" Version="$(SystemMemoryVersion)" />
```

Comment:
Condition was initially added in https://github.com/dotnet/roslyn/pull/26762/files#diff-864bb5e2533a421a81b2fc32ca13c208c8086e2ea9c3cd4f43d66cb719158c09L73-L74.
If it is still necessary for some reason, I think I should update them to net6.0 instead of removing.

I removed to preserve the current behavior where everything appears to be working properly.

cc @jaredpar

Reply:
I _think_ that this reference is only necessary when targeting .NET Framework at this point. It should come by default when we are using .NET Core. Can we condition this reference on .NET Framework? 

I agree this likely works as written. At the same time though it's unnecessarily pushing `System.Memory` to a version it would not be by default when targeting .NET Core. I worry about future headaches that may cause us. Simpler to just keep the defaults when targeting .NET Core

Reply:
@jaredpar I changed the condition for `System.Memory` and `System.Runtime.CompilerServices.Unsafe` to be on net472 only. For `System.IO.Pipes.AccessControl`, let me know what do you think. Currently its condition is always false and it's never included, so I deleted it completely. If it should be in a specific TFM, I'll fix that.

Reply:
IIRC `System.IO.Pipes.AccessControl` was needed in earlier .NET Core versions but included by default it later ones. If it compiles though then it's fine cause it was added for a very specific API. So either it's included by default now or we're jsut not using that API anymore.